### DISCUSSION TITLE
How can I add variables in my class?

### DISCUSSION THREAD

**Discussion by ntosis** [Pycord-Development/pycord#2083] [Q&A] [answered]
Hi, how can I extend the class with my variables? When I try the folloing code the the function, on message exits with the error "list with draft chabbels" is not defined, I tried also without self and with bot.list_xxx but always the same error. Any help?

class MyBot(commands.Bot):
    def __init__(self):
        self.list_with_draft_channels = [0]
        super().__init__(command_prefix=BOT_PREFIX, intents=discord.Intents.all()) #pass arguments to the parent class
        print("Mybot")

bot = MyBot()

@bot.event
async def on_message(message):
    print(self.list_with_draft_channels)

**Comment by JustaSqu1d** [comment, marked as answer]
To start off, the GitHub discussions tab is **not intended for help-related questions:** please ask in our [Discord server](https://discord.gg/pycord), instead.

Assuming that the indentation is correct, `self.list_with_draft_channels` will not work because  `self` is not defined. `self` is used to refer to the object of the creation, however, outside of the class, you would use the variable that refers to the object itself (in this case, `bot`).

As a result, you can use the object to access its attribute: `bot.list_with_draft_channels`.

**Comment by ntosis** [comment]
Hi ,

Thank you for your reply but I mentioned in my message that I have used the
bot.syntax but I have the same error that the class does not have an
element with the name list_xxx

**Reply by JustaSqu1d** [reply]
Hey, sorry for the ambiguity. I have attached an example of what my solution would look like.

```diff
class MyBot(commands.Bot):
    def __init__(self):
        self.list_with_draft_channels = [0]
        super().init(command_prefix=BOT_PREFIX, intents=discord.Intents.all()) #pass arguments to the parent class

print("Mybot")

bot = MyBot()

@bot.event
async def on_message(message):
-    print(self.list_with_draft_channels)
+    print(bot.list_with_draft_channels)
```

**Comment by ntosis** [comment]
I will try it out tomorrow.

**Comment by Lulalaby** [comment]
Please join our support server
